shopinvader_restapi/models/sale.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class SaleOrder(models.Model):
+    _name = "sale.order"
+    _inherit = ["sale.order", "track.external.mixin"]
+
+    typology = fields.Selection([("sale", "Sale"), ("cart", "Cart")], default="sale")
```

Comment:
This typology field is now defined in the `sale_cart` module, so should we depend on it to enable co-existence of the old and new cart apis? @lmignon @qgroulard @marielejeune .

Reply:
Agree. Then adapt the code in this module as `action_confirm_cart` and `action_confirm` are implemented in `sale_cart` too.